apps/sms/js/thread_ui.js

```diff
@@ -1864,7 +1859,10 @@ var ThreadUI = {
     }
 
     // Removing from DOM all messages to delete
+    var messages = Threads.active.messages;
     for (var i = 0, l = list.length; i < l; i++) {
+      var message = messages.get(+list[i]);
+      Threads.unregisterMessage(message);
```

Comment:
This looks not very neat since we will need to fetch the message by id everytime when we need to delete bunch of messages in same thread, but I'm not sure if it's necessary to have another method for this case.